src/Core/CSharpVB/RuleSetGenerator.cs

```diff
@@ -32,50 +32,40 @@
 //      the SLVS version is checked in, and will be updated as the QP changes. Ordering the ruleset
 //      reduces the churn when comparing old/new files.
 // * the SLVS version doesn't include taint-analysis rules
+// * changed to implement an interface for testability
 
 namespace SonarLint.VisualStudio.Core.CSharpVB
 {
-    public class RuleSetGenerator
+    public class RuleSetGenerator : IRuleSetGenerator
     {
         private const string SONARANALYZER_PARTIAL_REPO_KEY = "sonaranalyzer-{0}";
         private const string ROSLYN_REPOSITORY_PREFIX = "roslyn.";
 
-        private readonly IDictionary<string, string> sonarProperties;
         private readonly string inactiveRuleActionText = GetActionText(RuleAction.None);
 
         private readonly string activeRuleActionText = GetActionText(RuleAction.Warning);
 
-        public RuleSetGenerator(IDictionary<string, string> sonarProperties)
+        public RuleSet Generate(string language, IEnumerable<SonarQubeRule> rules, IDictionary<string, string> sonarProperties)
```

Comment:
Changed the method signature so the properties don't need to be passed in the constructor.